### PR TITLE
Fix observance logic for laborable holidays in Uruguay

### LINKED ISSUE #745: Some holidays of Uruguay are incorrect.
Hello.
Some holidays for Uruguay are incorrect.

- The Battle of stones was in May, 18.
- The Respect for Cultural Diversity Day or Columbus day is October, 12.

In Uruguay are 2 categories of holidays (laborables and not laborables). In the not laborable holidays nobody works while in the laborable holidays the work is optional.

The not laborables holidays are:
- 01/01
- 01/05
- 18/07
- 25/08
- 25/12

The rest of holidays are laborables, and depending of the day of the week, some of them are moved to earlier monday or to the next friday. The rule is:
- If the holiday is tuesday or wednesday, they moved to earlier monday.
- If the holiday is thursday, is moved to next friday.
- If the holiday is in monday or friday, doesn't move.

That is the reason of the error.
For example, the Battle of the Stones is the May, 18. In 2021 was tuesday and was moved to the earlier monday that was May, 17.

One exception to this rule is the November, 2 ("Día de los difuntos"). This holiday is laborable, but doesn't move.

Best regards.

### PR DESCRIPTION
**Pull request by kasya**
Change logic for 3 laborable holidays: Desembarco de los 33 Orientales en la playa de la Agraciada, Batalla de las Piedras and Día del Respeto a la Diversidad Cultural. 
Refactor code to group holidays by movable/unmovable logic and to satisfy DRY principle.

Resolves #745

### CODE REVIEW

File: holidays/countries/uruguay.py [hunk 1]
```diff
@@ -12,133 +12,99 @@
...
+                # [Respect for Cultural Diversity Day].
+                date(year, OCT, 12),
+                "Día del Respeto a la Diversidad Cultural "
+                "[Respect for Cultural Diversity Day]",
+            ),
+        )
+
+        for dt, name in holiday_pairs:
+            if dt.weekday() in (TU.weekday, WE.weekday):
+                self[dt - rd(weekday=MO(-1))] = name
```
Comment (arkid15r) [inline holidays/countries/uruguay.py:103]:
```suggestion
                self[dt + rd(weekday=MO(-1))] = name
```

File: test/countries/test_uruguay.py [hunk 2]
```diff
@@ -69,57 +47,135 @@ def test_labor_day(self):
...
+            self.assertEqual(self.holidays[dt], "Día de los difuntos")
 
-    def test_christmas(self):
-        for year in range(1900, 2100):
-            dt = date(year, 12, 25)
+    # Moveable holidays.
+
+    def test_carnival_day(self):
+        for dt in (
+            date(2021, 2, 15),
```
Comment (arkid15r) [inline test/countries/test_uruguay.py:110]: Sort here and below.

File: holidays/countries/uruguay.py [hunk 3]
```diff
@@ -12,133 +12,99 @@
...
-        if not self.observed and date(year, NOV, 2).weekday() in WEEKEND:
-            pass
-        else:
-            self[date(year, NOV, 2)] = name
-
-        # Christmas
-        self[date(year, DEC, 25)] = "Navidad [Christmas]"
+        # Holy Week.
+        self[
+            easter(year) - rd(weekday=TH(-1))
```
Comment (KJhellico) [inline holidays/countries/uruguay.py:72]:
```suggestion
            easter(year) + rd(weekday=TH(-1))
```

File: holidays/countries/uruguay.py [hunk 4]
```diff
@@ -12,133 +12,99 @@
...
+        ] = "Semana Santa (Jueves Santo)  [Holy day (Holy Thursday)]"
+        self[
+            easter(year) - rd(weekday=FR(-1))
```
Comment (KJhellico) [inline holidays/countries/uruguay.py:75]:
```suggestion
            easter(year) + rd(weekday=FR(-1))
```

File: holidays/countries/uruguay.py (hunk 1 above)
Comment (KJhellico) [inline holidays/countries/uruguay.py:103]:
```suggestion
                self[dt + rd(weekday=MO(-1))] = name
```

**Review by KJhellico — CHANGES_REQUESTED** [review]

**Comment by dr-prodigy** [comment]
Thank you very much @kasya ! 👍

**Pull request merged**